jsx/src/components/ServerDashboard/ServerDashboard.jsx

```diff
@@ -339,87 +346,65 @@ const ServerDashboard = (props) => {
                 </Button>
               </td>
             </tr>
-            {user_data.flatMap((e, i) => {
-              let userServers = Object.values({
-                "": e.server,
-                ...(e.servers || {}),
-              });
-              return userServers.map((server) => {
-                server = { name: "", ...server };
-                return (
-                  <tr key={i + "row"} className="user-row">
-                    {!server.name && (
-                      <td
-                        data-testid="user-row-name"
-                        rowspan={userServers.length}
-                      >
-                        {e.name}
-                      </td>
-                    )}
-                    {!server.name && (
-                      <td
-                        data-testid="user-row-admin"
-                        rowspan={userServers.length}
-                      >
-                        {e.admin ? "admin" : ""}
-                      </td>
-                    )}
+            {servers.map(([user, server], i) => {
+              server.name = server.name || "";
```

Comment:
Test seems to be [failing here](https://github.com/jupyterhub/jupyterhub/runs/5507233231?check_suite_focus=true#step:6:650) with 

```
    TypeError: Cannot read property 'name' of null
      348 |             </tr>
      349 |             {servers.map(([user, server], i) => {
    > 350 |               server.name = server.name || "";
          |                                    ^
      351 |               return (
      352 |                 <tr key={i + "row"} className="user-row">
      353 |                   <td data-testid="user-row-name">{user.name}</td>
```

Presumably need to handle `if server` here.

Reply:
It seems there are a lot of places below where `server.someProperty` is accessed, so addressing this requires several lines changes or to initialize some variable to a blank object.

I note that the `servers` list is created above, perhaps the sensible fix is to make sure it doesn't generate `null` entries for `server`. I've added a change suggestion.